pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -301,18 +300,37 @@ def sensitivity_calculation(method, instance, paramSubList, perturbList,
         
                     
         logger.info("ipopt starts")
-        ipopt.solve(m, tee=streamSoln)
+        ipopt.solve(m, tee=tee)
         m.ipopt_zL_in.update(m.ipopt_zL_out)  #: important!
         m.ipopt_zU_in.update(m.ipopt_zU_out)  #: important!    
         logger.debug("ipopt completed")
         #: k_aug
         logger.info("k_aug starts")
         kaug.options['dsdp_mode'] = ""  #: sensitivity mode!
-        kaug.solve(m, tee=streamSoln)
+        kaug.solve(m, tee=tee)
         logger.debug("k_aug completed")
         dotsens.options["dsdp_mode"] = ""
         logger.info("dotsens starts")
-        dotsens.solve(m, tee=streamSoln) 
+        dotsens.solve(m, tee=tee) 
         logger.debug("dotsens completed")
                 
     return m        
+
+def _add_kaug_suffix(model, suffix_name, _direction):
+    # _add_kaug_suffix checks the model to see if suffix_name already exists.
+    # It adds suffix_name to the model for a given direction '_direction'.
+    suffix_checker = model.component(suffix_name)
+    if _direction == 'IMPORT':
+        if suffix_checker is None:
+            setattr(model, suffix_name, Suffix(direction=Suffix.IMPORT))
+        else:
+            model.del_component(suffix_name)
+            setattr(model, suffix_name, Suffix(direction=Suffix.IMPORT))
```

Comment:
You don't want to just delete this, as the component was probably an important part of the model.  You want to use a method like is in GDP: https://github.com/Pyomo/pyomo/blob/c82264327400eaf2666b2fa637f0ec5e83e22ad2/pyomo/gdp/plugins/cuttingplane.py#L1038

Here, I would do something like:
```python
def _add_kaug_suffix(model, suffix_name, direction):
    suffix = rHull.component(suffix_name)
    if suffix is None:
        setattr(model, suffix_name, Suffix(direction=direction))
    else:
        if suffix.ctype is Suffix:
            return
        rHull.del_component(suffix)
        setattr(model, suffix_name, Suffix(direction=direction))
        rHull.add_component(unique_component_name(model, suffix_name), suffix)
```

Reply:
Thanks for the comments. I edited it like in GDP.